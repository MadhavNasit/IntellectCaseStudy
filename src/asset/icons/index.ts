export const iconRegistry = {
  // Add your new image path here..
};
