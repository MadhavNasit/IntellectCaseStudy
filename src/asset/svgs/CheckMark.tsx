import * as React from 'react';
import Svg, { SvgProps, Path } from 'react-native-svg';
import { memo } from 'react';
const CheckMark = (props: SvgProps) => (
  <Svg viewBox="0 0 512 512" {...props}>
    <Path
      fill="none"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={32}
      d="M416 128 192 384l-96-96"
    />
  </Svg>
);
const Memo = memo(CheckMark);
export default Memo;
