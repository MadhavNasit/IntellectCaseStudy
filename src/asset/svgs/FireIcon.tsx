import * as React from 'react';
import Svg, { SvgProps, Path } from 'react-native-svg';
import { memo } from 'react';
const FireIcon = (props: SvgProps) => (
  <Svg viewBox="0 0 512 512" {...props}>
    <Path
      fill="none"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={32}
      d="M112 320c0-93 124-165 96-272 66 0 192 96 192 272a144 144 0 0 1-288 0z"
    />
    <Path
      fill="none"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={32}
      d="M320 368c0 57.71-32 80-64 80s-64-22.29-64-80 40-86 32-128c42 0 96 70.29 96 128z"
    />
  </Svg>
);
const Memo = memo(FireIcon);
export default Memo;
