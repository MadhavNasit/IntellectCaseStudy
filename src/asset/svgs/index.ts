export { default as HomeIcon } from './HomeIcon';
export { default as HomeFilledIcon } from './HomeFilledIcon';
export { default as DiamondIcon } from './DiamondIcon';
export { default as DiamondFilledIcon } from './DiamondFilledIcon';
export { default as SettingsIcon } from './SettingsIcon';
export { default as SettingsFilledIcon } from './SettingsFilledIcon';
