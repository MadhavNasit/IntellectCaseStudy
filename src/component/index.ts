export * from './Icon';
export * from './Screen';
export * from './Text';
export * from './Button';
export * from './Greeting';
export { default as ProgressBar } from './ProgressBar';
export * from './Card';
export * from './CircularProgress';
