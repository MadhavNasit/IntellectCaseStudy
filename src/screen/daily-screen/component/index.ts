export * from './SessionCard';
export { default as SessionList } from './SessionList';
