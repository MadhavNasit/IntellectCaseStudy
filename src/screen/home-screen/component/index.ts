export * from './CircularInsightItem';
export * from './PersonalInsightCard';
