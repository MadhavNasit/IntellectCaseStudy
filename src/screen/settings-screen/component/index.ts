export * from './LanguageSelector';
export * from './ThemeSelector';
