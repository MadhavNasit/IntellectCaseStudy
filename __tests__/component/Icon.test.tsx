import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { Icon } from 'component';
import { BellIcon } from 'asset/svgs';

jest.mock('asset/icons', () => ({
  iconRegistry: {
    dailyBanner: require('asset/icons/daily-banner/daily-banner.png'),
  },
}));

describe('Icon Component', () => {
  it('renders correctly with default styles', () => {
    const { toJSON } = render(<Icon icon="dailyBanner" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders an image icon correctly', () => {
    const { getByTestId } = render(<Icon icon="dailyBanner" />);
    const image = getByTestId('icon-image');
    expect(image).toBeTruthy();
  });

  it('renders an SVG icon correctly', () => {
    const { getByTestId } = render(<Icon icon={<BellIcon />} />);
    expect(getByTestId('svg-icon')).toBeTruthy();
  });

  it('applies the provided color and size styles', () => {
    const { getByTestId } = render(
      <Icon icon="dailyBanner" color="red" size={30} />,
    );
    const image = getByTestId('icon-image');
    expect(image.props.style).toEqual(
      expect.arrayContaining([{ tintColor: 'red' }, { width: 30, height: 30 }]),
    );
  });

  it('triggers onPress when pressed', () => {
    const onPressMock = jest.fn();
    const { getByTestId } = render(
      <Icon icon="dailyBanner" onPress={onPressMock} />,
    );
    const touchable = getByTestId('icon-touchable');
    fireEvent.press(touchable);
    expect(onPressMock).toHaveBeenCalled();
  });

  it('renders inside a View when not pressable', () => {
    const { getByTestId } = render(<Icon icon="dailyBanner" />);
    expect(getByTestId('icon-view')).toBeTruthy();
  });

  it('renders correctly when provided with a custom style', () => {
    const { getByTestId } = render(
      <Icon icon="dailyBanner" style={{ margin: 10 }} />,
    );
    const container = getByTestId('icon-view');
    expect(container.props.style.margin).toBe(10);
  });
});
