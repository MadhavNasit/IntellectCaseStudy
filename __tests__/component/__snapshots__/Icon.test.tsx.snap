// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Component renders correctly with default styles 1`] = `
<View
  testID="icon-view"
>
  <Image
    source={
      {
        "testUri": "../../../src/asset/icons/daily-banner/daily-banner.png",
      }
    }
    style={
      [
        {
          "resizeMode": "contain",
        },
        false,
        false,
        undefined,
      ]
    }
    testID="icon-image"
  />
</View>
`;
